rust/xaynet-analytics/src/database/common.rs

```diff
@@ -1,35 +1,57 @@
 use anyhow::{anyhow, Error, Result};
 use isar_core::{
     index::StringIndexType,
-    object::{data_type::DataType, object_builder::ObjectBuilder},
+    object::{
+        data_type::DataType,
+        isar_object::{IsarObject, Property},
+        object_builder::ObjectBuilder,
+    },
     schema::collection_schema::CollectionSchema,
 };
-use std::vec::IntoIter;
+use std::{convert::From, vec::IntoIter};
 
 use crate::database::isar::IsarDb;
 
-pub trait IsarAdapter: Sized {
```

Comment:
Why it is `Sized`?

Reply:
without it it complains that the `Result<Self, Error>` from `read()` doesn't have a size known at compilation time

Reply:
Oh yes, didn't notice that.